src/renderer/components/ImageCropper/index.tsx

```diff
@@ -0,0 +1,284 @@
+
+    return [x, y]
+  }
+
+  useLayoutEffect(() => {
+    const onMouseDown = (ev: MouseEvent) => {
+      ev.preventDefault()
+
+      dragging.current = true
+
+      startX.current = ev.clientX
+      startY.current = ev.clientY
+    }
+
+    const onMouseUp = (ev: MouseEvent) => {
+      ev.preventDefault()
+
+      dragging.current = false
+
+      posX.current +=
+        (ev.clientX - startX.current) * Math.min(1 / zoom.current, 1)
+      posY.current +=
+        (ev.clientY - startY.current) * Math.min(1 / zoom.current, 1)
+      moveImages(posX.current, posY.current)
+    }
+
+    const onMouseMove = (ev: MouseEvent) => {
+      if (!dragging.current) {
+        return
+      }
+      // we never go faster with zoom above 1, but we speed up zoomed-out (below 1) by inverting the zoom value
+      const dX = (ev.clientX - startX.current) * Math.min(1 / zoom.current, 1)
+      const dY = (ev.clientY - startY.current) * Math.min(1 / zoom.current, 1)
+
+      moveImages(posX.current + dX, posY.current + dY)
+    }
+
+    const onZoom = (ev: WheelEvent) => {
+      // negate delta so when we wheel forward we zoom in
+      zoom.current *= Math.exp((ev.deltaY > 1 ? -1 : 1) * 0.2)
+
+      moveImages(posX.current, posY.current)
+    }
+
+    document.addEventListener('mousemove', onMouseMove)
+    document.addEventListener('mousedown', onMouseDown)
+    document.addEventListener('mouseup', onMouseUp)
+    document.addEventListener('wheel', onZoom)
+
+    return () => {
+      document.removeEventListener('mousemove', onMouseMove)
+      document.removeEventListener('mousedown', onMouseDown)
+      document.removeEventListener('mouseup', onMouseUp)
+      document.removeEventListener('wheel', onZoom)
+    }
+  })
+  return (
+    <Dialog
+      className={styles.imageCropperDialog}
+      canEscapeKeyClose
+      onClose={onClose}
+      canOutsideClickClose={false}
+    >
+      <DialogHeader></DialogHeader>
+      <DialogBody className={styles.imageCropperDialogBody}>
+        <DialogContent>
+          <div
+            ref={container}
+            style={{
+              height: '500px',
+            }}
+            className={styles.imageCropperContainer}
+          >
+            <div ref={shade} className={styles.imageCropperShade}></div>
+            <img
+              ref={cutImage}
+              className={styles.imageCropperCutImage}
+              src={filepath}
+              onLoad={setupImages}
+            />
+            <img
+              ref={fullImage}
+              className={styles.imageCropperFullImage}
+              src={filepath}
+            />
+          </div>
+        </DialogContent>
+      </DialogBody>
+      <DialogFooter>
+        <FooterActionButton
+          onClick={() => {
+            onCancel()
+            onClose()
+          }}
+        >
+          {tx('cancel')}
+        </FooterActionButton>
+        <FooterActionButton onClick={onSubmit}>{tx('save')}</FooterActionButton>
```

Comment:
The `FooterActionButton` needs to be wrapped in `FooterActions` for correct layout, check other dialog implementations for reference